frontend/src/metabase/ui/components/inputs/Switch/Switch.styled.tsx

```diff
@@ -74,8 +74,10 @@ export const getSwitchOverrides = (): MantineThemeOverride["components"] => ({
           color: theme.fn.themeColor("text-dark"),
           cursor: "pointer",
           "&[data-disabled]": {
-            color: theme.fn.themeColor("text-light"),
-            cursor: "default",
+            "&:not([data-css-specificity-hack='😢'])": {
```

Comment:
I don't see any reference to this specificity hack in the codebase. Is this doing some magic with `:not()` or do we need to actually set the `data-css-specificity-hack` attribute on our elements?

Reply:
yeah, adding the `:not` gives this block a higher specificity to elevate it above the built in mantine style that is setting the text to transparent. it's a hack, I don't like it, but I like that it's clearly a hack. I'm hopeful that this can be removed in mantine v7

Reply:
this is just to win the specificity from the rule that sets the color to `transparent`. I understand that we don't need to set anything here.

Reply:
If it's a hack, can we document that in a comment?

Reply:
like ?

```ts
// this is a hack to raise the css specificity
"&[data-disabled]:not([data-css-specificity-hack='😢'])" {
```

seems redundant?

Reply:
I think the code itself is sufficient in this case.